### PR TITLE
Update PrivateEndpoints to new child-resource structure

### PR DESCRIPTION
**Pull request by MariusStorhaug**
# Change

[Network: Privateendpoints](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.privateendpoints.yml)

- Updated Private EndPoints resource and child-resources to new child resource structure

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: arm/Microsoft.Network/privateEndpoints/privateDnsZoneGroups/deploy.bicep [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    privateDnsZoneConfigs: privateDnsZoneConfigs
+  }
+}
+
+@description('The name of the private endpoint DNS zone group')
+output privateDnsZoneGroupName string = privateDnsZoneGroup.name
+
+@description('The resourceId of the private endpoint DNS zone group')
+output privateDnsZoneGroupResourceId string = privateDnsZoneGroup.id
+@description('The resource group the private endpoint DNS zone group was deployed into')
```
Comment (eriqua) [inline arm/Microsoft.Network/privateEndpoints/privateDnsZoneGroups/deploy.bicep:26]: Minor: missing new line

File: arm/Microsoft.Network/privateEndpoints/deploy.bicep [hunk 2]
```diff
@@ -11,7 +11,7 @@ param serviceResourceId string
 param groupId array
 
 @description('Optional. Resource id of the private DNS zone.')
```
Comment (eriqua) [inline arm/Microsoft.Network/privateEndpoints/deploy.bicep:13]: Minor: should we update the param description to reflect the change?

File: arm/Microsoft.Network/privateEndpoints/deploy.bicep [hunk 3]
```diff
@@ -91,6 +77,14 @@ module privateEndpoint_rbac '.bicep/nested_rbac.bicep' = [for (roleAssignment, i
   }
 }]
 
+module privateEndpoint_privateDnsZoneGroups 'privateDnsZoneGroups/deploy.bicep' = [for (privateDnsZoneGroup, index) in privateDnsZoneGroups: {
```
Comment (eriqua) [inline arm/Microsoft.Network/privateEndpoints/deploy.bicep:80]: More a question: shoul we keep child modules close to the parents and have parent providers after them?

**Review by eriqua — COMMENTED** [review]
An additional general comment if that makes sense: should we add a min.parameters.json also in this case to test the minimum set of required parameters without children?

File: arm/Microsoft.Network/privateEndpoints/deploy.bicep (hunk 3 above)
Comment (MariusStorhaug) [reply arm/Microsoft.Network/privateEndpoints/deploy.bicep]: Put it closer to the parent object. Just though that children would be after describing the whole main component, i.e. rbac, diag and locks as they are scoped to the parent resource.

File: arm/Microsoft.Network/privateEndpoints/deploy.bicep (hunk 2 above)
Comment (MariusStorhaug) [reply arm/Microsoft.Network/privateEndpoints/deploy.bicep]: Fixed.

File: arm/Microsoft.Network/privateEndpoints/privateDnsZoneGroups/deploy.bicep (hunk 1 above)
Comment (MariusStorhaug) [reply arm/Microsoft.Network/privateEndpoints/privateDnsZoneGroups/deploy.bicep:27]: Fixed.

**Review by eriqua — APPROVED** [review]
🆗
